src/App.tsx

```diff
@@ -15,71 +15,86 @@ export const goods = [
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  onSelect = (selectedGood: string) => {
+    this.setState({ goodSelected: selectedGood });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render(): JSX.Element {
+    const { goodSelected } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+    return (
+      <main className="section container">
+        {goodSelected ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${goodSelected} is selected`}
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
+              aria-label="remove"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              className="delete ml-3"
+              onClick={this.clearSelected}
+            />
+          </h1>
+        ) : (
+          <h1 className="title">No goods selected</h1>
+        )}
+        <table className="table">
+          <tbody>
+            {goods.map((good) => (
+              goodSelected === good ? (
+                <tr
+                  data-cy="Good"
+                  className="has-background-success-light"
+                  key={good}
+                >
+                  <td>
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={this.clearSelected}
+                    >
+                      -
+                    </button>
+                  </td>
+                  <td data-cy="GoodTitle" className="is-vcentered">
+                    {good}
+                  </td>
+                </tr>
+              ) : (
+                <tr data-cy="Good" key={good}>
+                  <td>
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => this.onSelect(good)}
+                    >
+                      +
+                    </button>
+                  </td>
+                  <td data-cy="GoodTitle" className="is-vcentered">
+                    {good}
+                  </td>
+                </tr>
```

Comment:
You can use a slightly different approach:
```
{goods.map(good => (
  <tr
    data-cy="Good"
    key={good}
    className={goodSelected === good ? "has-background-success-light" : ""}
  >
    <td>
      {goodSelected === good ? (
        <button
          // all props for the selected good button
        />
      ) : (
        <button
          // all props for the add button
        />
      )}
    </td>
  </tr>
```
So, your code will be shorter.

Reply:
Yes, that makes sense, thank you!